### PR TITLE
Tempo: Separate trace to logs and loki search datasource config

### PR DESCRIPTION
**Pull request by connorlindsey**
**What this PR does / why we need it**:
We currently use the linked Trace to Logs datasource for both trace to logs and loki search. Trace to logs will soon support other logs datasources (like Splunk) so we need to separate the config. This PR adds new config to Tempo and adds fallback logic to use the correct datasource. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/tempo/datasource.ts [hunk 1]
```diff
@@ -100,11 +107,18 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
       return of({ data: [], state: LoadingState.Done });
     }
 
+    // Get linked logs datasource. Fall back to legacy loki search/trace to logs config
+    const legacyLogsDatasourceUid =
+      this.tracesToLogs?.lokiSearch !== false && this.lokiSearch === undefined
+        ? this.tracesToLogs?.datasourceUid
+        : undefined;
```
Comment (joey-grafana) [inline public/app/plugins/datasource/tempo/datasource.ts:114]: Would be cool to extract this to util method and share with line 70-73 in QueryField.

**Review by joey-grafana — COMMENTED** [review]
Seems to work as expected, although I didn't have this previously setup, so I didn't have the legacy option selected. Could somebody please double check this for Connor and myself?

Sorry, it's getting late into the evening in Ireland and I'm falling asleep zzz

Thanks for the TNS tip btw! 

Loki config
Tempo config
Loki search

File: docs/sources/datasources/tempo.md [hunk 2]
```diff
@@ -57,6 +57,12 @@ This is a configuration for the beta Node Graph visualization. The Node Graph is
 
 -- **Enable Node Graph -** Enables the Node Graph visualization.
 
+### Loki Search
+
+This is a configuration for the Loki search query type.
+
+-- **Data source -** Loki instance used when searching for traces. The Loki instance must have derived fields configured.
```
Comment (chri2547) [inline docs/sources/datasources/tempo.md:64]:
```suggestion
-- **Data source -** The Loki instance in which you want to search traces. You must configure derived fields in the Loki instance.
```

**Review by chri2547 — CHANGES_REQUESTED** [review]
Slightly adjusted wording. Please review to see if my suggestion makes sense. Thanks!

**Review by CatPerry — APPROVED** [review]

**Review by chri2547 — APPROVED** [review]
Approving so that I unblock the merge.

File: docs/sources/datasources/tempo.md (hunk 2 above)
Comment (connorlindsey) [reply docs/sources/datasources/tempo.md]: Committing in the UI isn't working right now, so I'll manually add in those changes

**Pull request merged**